test/sharness/t0114-gateway-subdomains.sh

```diff
@@ -904,6 +904,24 @@ test_hostname_gateway_response_should_contain \
   "http://127.0.0.1:$GWAY_PORT/" \
   "$CID_VAL"
 
+## ============================================================================
+## Test support for overriding implicit defaults
+## ============================================================================
+
+# disable subdomain gateway at localhost by removing implicit config
+ipfs config --json Gateway.PublicGateways '{
+  "localhost": null
+}' || exit 1
```

Comment:
Added explicit test that ensures users are able to disable subdomain gw on `localhost` if they wish to do so.